hubblestack/files/hubblestack_nova/misc.py

```diff
@@ -509,6 +509,256 @@ def check_core_dumps(reason=''):
+    ungrouped_partition_files = ungrouped_partition_files.split('\n') if ungrouped_partition_files != "" else []
+    ungrouped_files = ungrouped_files + ungrouped_partition_files
+    if ungrouped_files == []:
+        return True
+
+    return str(list(set(ungrouped_files)))
+
+
+def check_all_users_home_directory(reason=''):
+    '''
+    Ensure all users' home directories exist
+    '''
+
+    users_uids_dirs = _execute_shell_command("cat /etc/passwd | awk -F: '{ print $1 \" \" $3 \" \" $6 }'").strip()
+    users_uids_dirs = users_uids_dirs.split('\n') if users_uids_dirs != "" else []
+    error = []
+    for user_data in users_uids_dirs:
+        user_uid_dir = user_data.strip().split(" ")
+        if len(user_uid_dir) < 3:
+                user_uid_dir = user_uid_dir + ['']*(3-len(user_uid_dir))
+        if user_uid_dir[1].isdigit():
+            if int(user_uid_dir[1]) >= 1000 and not os.path.isdir(user_uid_dir[2]) and user_uid_dir[0] is not "nfsnobody":
+                error += ["The home directory " + user_uid_dir[2] + " of user " + user_uid_dir[0] + " does not exist."]
+        else:
+            error += ["User " + user_uid_dir[0] + " has invalid uid " + user_uid_dir[1]]
+    if error == []:
+        return True
+
+    return str(error)
+
+
+def check_users_home_directory_permissions(reason=''):
+    '''
+    Ensure users' home directories permissions are 750 or more restrictive
+    '''
+
+    users_dirs = _execute_shell_command("cat /etc/passwd | egrep -v '(root|halt|sync|shutdown)' | awk -F: '($7 != \"/sbin/nologin\") {print $1\" \"$6}'").strip()
+    users_dirs = users_dirs.split('\n') if users_dirs != "" else []
+    error = []
+    for user_dir in users_dirs:
+        user_dir = user_dir.split(" ")
+        if len(user_dir) < 2:
+                user_dir = user_dir + ['']*(2-len(user_dir))
+        if user_dir[1] is None or user_dir[1] == "":
+            error += ["User " + user_dir[0] + " does not have any home directoyr"]
+        else:
+            result = restrict_permissions(user_dir[1], "750")
+            if result is not True:
+                error += ["permission on home directory " + user_dir[1]  + " of user " + user_dir[0] + " is wrong: " + result]
+
+    if error == []:
+        return True
+
+    return str(error)
+
+
+def check_users_own_their_home(reason=''):
+    '''
+    Ensure users own their home directories
+    '''
+
+    users_uids_dirs = _execute_shell_command("cat /etc/passwd | awk -F: '{ print $1 \" \" $3 \" \" $6 }'").strip()
+    users_uids_dirs = users_uids_dirs.split('\n') if users_uids_dirs != "" else []
+    error = []
+    for user_data in users_uids_dirs:
+        user_uid_dir = user_data.strip().split(" ")
+        if len(user_uid_dir) < 3:
+                user_uid_dir = user_uid_dir + ['']*(3-len(user_uid_dir))
+        if user_uid_dir[1].isdigit():
+            if int(user_uid_dir[1]) >= 1000 and os.path.isdir(user_uid_dir[2]) and user_uid_dir[0] is not "nfsnobody":
```

Comment:
On some other OS like Amazon Linux, we need value as 500 rather than 1000, can you please take this as parameter as max_system_uid and use it here. So that we can use this function for Amazon Linux as well in future.